### ISSUE TITLE
Exception while upgrading the amplify cli.

### ISSUE THREAD

**Issue by ecc7220** [aws-amplify/amplify-cli#10595]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v16.14.0

### Amplify CLI Version

8.5.0, 8.4.0, 8.3.0

### What operating system are you using?

Amazon Linux 2 on Cloud9 instance.

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made.

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

I'm getting lately this error when I upgrade the amplify cli:

`Developer2:~/environment/xxxxx (xxxxx) $ amplify upgrade
Downloading latest Amplify CLI
100% [========================================] 0.0 seconds left
⠸ Validating downloaded Amplify CLI....internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module '/home/ec2-user/environment/xxxx/--version'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function._resolveFilename (pkg/prelude/bootstrap.js:1819:46)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Function.runMain (pkg/prelude/bootstrap.js:1847:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
✖ An unexpected error has occurred, opt in to send an error report to AWS Amplify with non-sensitive project configuration files. Confirm  (y/N)`

The last line is new, it was first stuck showing waiting for validation, after a control-c, the "An unexpected..." message appeared, but I could not select the confirm, the script was not running anymore.

The version is now 8.5.0, prior was 8.4.0, which had the same error, but the version seems to work regardless of the exception.

It's kind of annoying though, I always need to cross fingers when I upgrade the cli. 

### Expected behavior

Clean upgrade with no issues.

### Reproduction steps

I can't provide steps to reproduce, no time currently to do so.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
2022-06-14T15:01:53.686Z|info : amplify upgrade core  
2022-06-14T15:01:53.736Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2022-06-14T15:06:52.247Z|info : amplify version core  {"version":true,"yes":false}
2022-06-14T15:07:02.753Z|info : amplify status core  {"yes":false}
2022-06-14T15:07:02.806Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2022-06-14T15:07:04.511Z|info : amplify-provider-awscloudformation.display-helpful-urls.showCognitoSandBoxMessage(["userPool"])


```

</details>

### Additional information

The log does not show any errors, see above.

**Comment by ecc7220** [comment]
I noticed, that I'm getting the following weird messages on push with the new version 8.5.0, there where apparently fixed in 8.4.0, but appeared also on 8.3.0 as I remember.

```Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
Overrides functionality is not implemented for this category
✔ Successfully pulled backend environment staging from the cloud.
Overrides functionality is not implemented for this categor```

**Comment by josefaidt** [comment]
Hey @ecc7220 :wave: thanks for raising this! This is a duplicate bug of https://github.com/aws-amplify/amplify-cli/issues/10583#issuecomment-1154508064

> There was a bug in `amplify upgrade` in amplify-cli 8.4.0 which has been fixed in amplify-cli 8.5.0 . Please install the CLI using `npm install -g @aws-amplify/cli@8.5.0` . The problem in amplify upgrade will not be seen in future versions of the CLI. Closing this, since resolved in 8.5.0.

And for your second note this is documented as an active bug here https://github.com/aws-amplify/amplify-cli/issues/10546

Closing in favor of the explanation on #10583 and duplicate bug of #10546
